packages/orbit-components/src/Toast/types.d.ts

```diff
@@ -44,15 +44,19 @@ export interface Toast {
 }
 
 export interface Options<T> {
-  readonly icon: Pick<DefaultToastOptions, "icon">;
   readonly loading: Renderable;
   readonly success: ValueOrFunction<Renderable, T>;
   readonly error: ValueOrFunction<Renderable, any>;
 }
 
-export type createToast = (
-  message: ValueOrFunction<Renderable, ToastType>,
-  options?: Pick<DefaultToastOptions, "icon">,
-) => void;
+export type IconType = Pick<DefaultToastOptions, "icon">;
 
-export type createToastPromise = <T>(promise: Promise<T>, options: Options<T>) => Promise<T>;
+export type createToast = (message: ValueOrFunction<Renderable, Toast>, options?: IconType) => void;
+
+export type createToastPromise = <T>(
+  promise: Promise<T>,
+  messages: Options<T>,
+  options?: IconType & { [key in ToastType]?: IconType },
```

Comment:
```suggestion
  options?: IconType & Record<ToastType, IconType>
```

Reply:
Partial<Record<<ToastType>, IconType>> can work then

Reply:
Right, they're the same. But I'll switch to your suggestion to make the code more Typescript conventional 👍